### PR TITLE
RavenDB-22190 - use the verifier from BouncyCastle

### PR DESCRIPTION
**Pull request by grisha-kotler**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-22190

### Additional description

Use the signature verifier from `BouncyCastle`.

### Type of change

- [ ] Bug fix
- [ ] Regression bug fix
- [x] Optimization
- [ ] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [x] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

**Review by ayende — COMMENTED** [review]
Good solution for this, yes.

File: src/Raven.Server/Commercial/LicenseValidator.cs [hunk 1]
```diff
@@ -87,21 +91,21 @@ private static byte[] GetBytesFromBase64String(string str)
                 ms.SetLength(attributesLen);
 
                 using (var binaryWriter = new BinaryWriter(ms))
-                using (var rsa = RSA.Create())
                 {
                     binaryWriter.Write(licenseKey.Id.ToByteArray());
                     binaryWriter.Write(licenseKey.Name);
 
-                    rsa.ImportParameters(rsaParameters);
+                    RsaKeyParameters publicKey = new(false,
```
Comment (ppekrol) [inline src/Raven.Server/Commercial/LicenseValidator.cs]: can you use named param? It is hard to understand in GH what this 'false' is for

File: src/Raven.Server/Commercial/LicenseValidator.cs [hunk 2]
```diff
@@ -87,21 +91,21 @@ private static byte[] GetBytesFromBase64String(string str)
...
+                        new BigInteger(1, rsaParameters.Modulus),
+                        new BigInteger(1, rsaParameters.Exponent));
 
-                    using (var sha1 = SHA1.Create())
-                    {
-                        ms.Position = 0;
-                        var hash = sha1.ComputeHash(ms);
+                    var dataToVerify = ms.ToArray();
+                    ISigner verifier = SignerUtilities.GetSigner("SHA1withRSA");
+                    verifier.Init(false, publicKey);
```
Comment (ppekrol) [inline src/Raven.Server/Commercial/LicenseValidator.cs]: named param

**Review by ppekrol — APPROVED** [review]

File: src/Raven.Server/Commercial/LicenseValidator.cs (hunk 1 above)
Comment (grisha-kotler) [reply src/Raven.Server/Commercial/LicenseValidator.cs]: done

File: src/Raven.Server/Commercial/LicenseValidator.cs (hunk 2 above)
Comment (grisha-kotler) [reply src/Raven.Server/Commercial/LicenseValidator.cs]: done

**Review by ppekrol — APPROVED** [review]

**Review by ArieSLV — APPROVED** [review]
I built this branch and tested it on `Rocky Linux 9.3` with the default restrictions that prohibit the use of `SHA-1`.

I confirm that this fix works in this environment.

**Pull request merged**